crates/rpc/src/cairo/ext_py/types.rs

```diff
@@ -0,0 +1,65 @@
+use serde::{Deserialize, Serialize};
+use serde_with::serde_as;
+use stark_hash::Felt;
+
+use crate::v03::method::simulate_transaction::dto::{
+    Address, CallType, EntryPointType, Event, MsgToL1,
+};
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct TransactionSimulation {
+    pub trace: TransactionTrace,
+    pub fee_estimation: FeeEstimate,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct TransactionTrace {
+    pub validate_invocation: Option<FunctionInvocation>,
+    pub function_invocation: Option<FunctionInvocation>,
+    pub fee_transfer_invocation: Option<FunctionInvocation>,
+    pub signature: Vec<Felt>,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct FunctionInvocation {
+    pub calldata: Vec<Felt>,
+    pub contract_address: Address,
+    pub selector: Felt,
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Option::is_none")]
```

Comment:
nit: If there's no flattening in/around these structs then you can simply use `serde_with::skip_serializing_none`